### DISCUSSION TITLE
Potential memory leak

### DISCUSSION THREAD

**Discussion by laggarcia** [kewisch/gdata-provider#502] [Support] [answered]
Hello,

First and foremost, thank you very much for the work on the Provider for Google Calendar Thunderbird add-on. I have been using it for many many years (I don't even recall when I started). Thank you very much!

After upgrading to the most recent versions (102.0.x), whenever the add-on starts to sync my calendars, the amount of memory being used by Thunderbird starts to increase rapidly. I have 48 GB of RAM, and even if I have nothing else running apart from my Fedora 36 booted and Thunderbird, the Linux kernel kills the Thunderbird process because it consumes all the available free memory.

Now, I have to say I am a really hard user (I suppose). I have more than 70 calendars being synced, and a few of them have thousands of events. However, this has been my situation for pretty much the whole year, and I just started noticing the out-of-memory issue since the last Thunderbird and add-on upgrade.

I wasn't able to find any description like this in the issues or the FAQ or the discussion forum. Has this been noticed somewhere else? I would be glad to help debugging the issue if it is the case.

**Comment by kewisch** [comment, marked as answer]
I don't think this is something I can fix in the provider. I'm not caching anything that might grow, and the code is just JavaScript. Not to say it can't be the Provider, but I rather suspect it might be something in Thunderbird. Can you try filing an issue there?
